waspc/data/Generator/templates/react-app/src/test/vitest/helpers.tsx

```diff
@@ -0,0 +1,55 @@
+import React from 'react'
+import { rest } from 'msw'
+import { setupServer } from 'msw/node'
+import { BrowserRouter as Router } from 'react-router-dom'
+import { render, RenderResult, cleanup } from '@testing-library/react'
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query'
+import { beforeAll, afterEach, afterAll } from 'vitest'
+import { Query } from '../../queries'
+import config from '../../config'
+
+export function renderInContext(ui: React.ReactElement): RenderResult {
+  const client = new QueryClient()
+  const { rerender, ...result } = render(
```

Comment:
@shayneczyzewski Why are we repeating `QueryClientProvider` twice in this file?

Reply:
If we have a link that points to the docs where they recommend this, I suggest adding it to the code.

Reply:
It was mostly random people on SO or in GH discussions. I can at least point to that code as a reference in the comment. 👍🏻